src/Context/Context.js

```diff
@@ -11,6 +11,17 @@ const FilterContext = createContext({});
 
 const FilterContextProvider = ({ children }) => {
   const [products, setProducts] = useState([]);
+  const [userLoggedIn, setUserLoggedIn] = useState(false);
+  useEffect(() => {
+    // console.log(sessionStorage.getItem("token"));
+    if (sessionStorage.getItem("token") === null) {
+      setUserLoggedIn(false);
+    } else if (sessionStorage.getItem("token") === "undefined") {
+      setUserLoggedIn(false);
+    } else {
+      setUserLoggedIn(true);
```

Comment:
this is correct but , i think you dont have to check for both  try this  ,( if am wrong plz correct me  ) 
if ( ! sessionStorage.getItem("token") ) { 
setUserLoggedIn(false);
}else{
setUserLoggedIn(true);
}